portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,42 +26,59 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-    private List<String> messages;
+    
+    public class Name 
+    {
+        private ArrayList<String> names = new ArrayList<String>();
+    }
   
-  @Override
-  public void init() 
-  {
-    messages = new ArrayList<>();
-    messages.add("This is message one.");
-    messages.add("This is messsage two! Welcome to the party json!");
-    messages.add("This is messsage three! This is too easy...");
-  }
+    private Name name = new Name();
+
+  
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Gson gson = new Gson(); 
+        String json = gson.toJson(name);
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        getNames(request);
+        response.sendRedirect("https://8080-dot-10831547-dot-devshell.appspot.com/");
```

Comment:
Why are we redirecting here? You should be able to just send a response that the JavaScript engine can use on the other end. This will cause the page to reload which isn't what we want.